test/spmd/test_inference_spmd_dynamo_imagenet.py

```diff
@@ -0,0 +1,238 @@
+    '--use_dynamo': {
+        'action': 'store_true',
+    },
+}
+
+FLAGS = args_parse.parse_common_options(
+    datadir='/tmp/imagenet',
+    profiler_port=9012,
+    opts=MODEL_OPTS.items(),
+)
+
+import os
+import numpy as np
+import time
+import torch
+import torchvision
+import torchvision.transforms as transforms
+import torch_xla
+import torch_xla.debug.metrics as met
+import torch_xla.distributed.parallel_loader as pl
+import torch_xla.utils.utils as xu
+import torch_xla.core.xla_model as xm
+import torch_xla.debug.profiler as xp
+import torch_xla.test.test_utils as test_utils
+import torch_xla.experimental.xla_sharding as xs
+import torch_xla.experimental.pjrt as pjrt
+
+DEFAULT_KWARGS = dict(
+    target_latency=0.0005,  # 0.50 ms
+)
+
+# Set any args that were not explicitly given by the user.
+default_value_dict = DEFAULT_KWARGS
+for arg, value in default_value_dict.items():
+  if getattr(FLAGS, arg) is None:
+    setattr(FLAGS, arg, value)
+
+
+def get_model_property(key):
+  default_model_property = {
+      'img_dim': 224,
+      'model_fn': getattr(torchvision.models, FLAGS.model)
+  }
+  model_properties = {
+      'inception_v3': {
+          'img_dim': 299,
+          'model_fn': lambda: torchvision.models.inception_v3(aux_logits=False)
+      },
+  }
+  model_fn = model_properties.get(FLAGS.model, default_model_property)[key]
+  return model_fn
+
+
+def inference_imagenet():
+  print('==> Preparing data..')
+  img_dim = get_model_property('img_dim')
+  device = xm.xla_device()
+  if FLAGS.fake_data:
+    assert FLAGS.test_set_batch_size == 1
+    test_loader = xu.SampleGenerator(
+        data=(torch.zeros(FLAGS.test_set_batch_size, 3, img_dim,
+                          img_dim).to(device),
+              torch.zeros(FLAGS.test_set_batch_size,
+                          dtype=torch.int64).to(device)),
+        sample_count=FLAGS.sample_count // FLAGS.test_set_batch_size)
+  else:
+    normalize = transforms.Normalize(
+        mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225])
+    resize_dim = max(img_dim, 256)
+    test_dataset = torchvision.datasets.ImageFolder(
+        os.path.join(FLAGS.datadir, 'val'),
+        # Matches Torchvision's eval transforms except Torchvision uses size
+        # 256 resize for all models both here and in the train loader. Their
+        # version crashes during training on 299x299 images, e.g. inception.
+        transforms.Compose([
+            transforms.Resize(resize_dim),
+            transforms.CenterCrop(img_dim),
+            transforms.ToTensor(),
+            normalize,
+        ]))
+
+    # For single-host SPMD, no data sampler is needed.
+    test_loader = torch.utils.data.DataLoader(
+        test_dataset,
+        batch_size=FLAGS.test_set_batch_size,
+        sampler=None,
+        drop_last=FLAGS.drop_last,
+        shuffle=False)
+
+  torch.manual_seed(42)
+
+  model = torchvision.models.resnet50().to(
+      device)  # get_model_property('model_fn')().to(device)
```

Comment:
Need to clean this up, either use `get_model_property('model_fn')().to(device)` or stick to `resnet50()`, in which case we don't need some of the model props code above.